src/main/scala/com/bwsw/cloudstack/vault/server/cloudstack/util/ApacheCloudStackTaskWrapper.scala

```diff
@@ -0,0 +1,79 @@
+package com.bwsw.cloudstack.vault.server.cloudstack.util
+
+import java.util.UUID
+
+import br.com.autonomiccs.apacheCloudStack.client.{ApacheCloudStackClient, ApacheCloudStackRequest}
+import br.com.autonomiccs.apacheCloudStack.client.beans.ApacheCloudStackUser
+import com.bwsw.cloudstack.vault.server.cloudstack.entities.Tag
+import com.bwsw.cloudstack.vault.server.util.{ApplicationConfig, ConfigLiterals}
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Created by medvedev_vv on 21.08.17.
+  */
+class ApacheCloudStackTaskWrapper {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+  private val secretKey = ApplicationConfig.getRequiredString(ConfigLiterals.cloudStackSecretKey)
+  private val apiKey = ApplicationConfig.getRequiredString(ConfigLiterals.cloudStackApiKey)
+  private val apacheCloudStackUser = new ApacheCloudStackUser(secretKey, apiKey)
+  private val cloudStackUrlList: Array[String] = ApplicationConfig
+    .getRequiredString(ConfigLiterals.cloudStackApiUrlList)
+    .split("[,\\s]+")
+  private val apacheCloudStackClientList = cloudStackUrlList.map { x =>
+    new ApacheCloudStackClient(x, apacheCloudStackUser)
+  }.toList
+
+  apacheCloudStackClientList.foreach(_.setValidateServerHttpsCertificate(false))
+
+  private var threadLocalClientList: ThreadLocal[List[ApacheCloudStackClient]] = new ThreadLocal
+  threadLocalClientList.set(apacheCloudStackClientList)
+
+  def getTagTask(resourceType: String, resourceId: UUID): () => String = {
+    val tagRequest = new ApacheCloudStackRequest("listTags")
+    tagRequest.addParameter("response", "json")
+    tagRequest.addParameter("resourcetype", resourceType)
+    tagRequest.addParameter("listAll", "true")
+    tagRequest.addParameter("resourceid", resourceId)
+
+    executeRequest(tagRequest, s"get tag by resourse: ($resourceId, $resourceType)")
+  }
+
+  def getEntityTask(parameterValue: String, parameterName: String, command: String): () => String = {
+    val request = new ApacheCloudStackRequest(command)
+    request.addParameter("response", "json")
+    request.addParameter("listAll", "true")
+    request.addParameter(parameterName, parameterValue)
+
+    executeRequest(request, s"get entity by command: $command")
+  }
+
+  def setTagToResourseTask(tag: Tag, resourseId: UUID, resourseType: String): () => String = {
+    val request = new ApacheCloudStackRequest("createTags")
+    request.addParameter("response", "json")
+    request.addParameter("resourcetype", resourseType)
+    request.addParameter("resourceids", resourseId)
+    request.addParameter("tags[0].key", tag.key)
+    request.addParameter("tags[0].value", tag.value)
+
+    executeRequest(request, s"set tags to resourse: ($resourseId, $resourseType)")
+  }
+
+  private def executeRequest(request: ApacheCloudStackRequest, requestDescription: String)(): String = {
```

Comment:
executeRequet -> createRequest (method doesn't execute a request and just create lambda to execute a request later)